### PR TITLE
e2eTests: Adds cleanup of created datasource and dashboard

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Add cleanup of created datasource and dashboard during tests so you can run the e2e tests several times without creating massive amounts of test data sources and test dashboards.
Also some minor refactor like not running image compare locally because that will almost always break as the `truth image` is taken from CircleCI output and because not everyone has PhantomJS or imagerenderer working locally.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
LGTM!

In the future it wold be great to have a way to easily extend this for plugin datasources -- that may be just copy/paste, or perhaps just defining what needs set on the config page

**Comment by hugohaggmark** [comment]
> LGTM!
> 
> In the future it wold be great to have a way to easily extend this for plugin datasources -- that may be just copy/paste, or perhaps just defining what needs set on the config page

I was thinking of evaluating a couple of other e2e frameworks before doing anymore changes to this home grown one. But I think you can "easily" extend that datasource page interface with the specific plugins fields. Your just need to remember to add `aria-label` selectors to your plugin.

**Comment by ryantxu** [comment]
Yes we should totally evaluate various options before copying the pattern everywhere!

File: public/app/features/datasources/settings/DataSourceSettingsPage.tsx [hunk 1]
```diff
@@ -3,36 +3,31 @@ import React, { PureComponent } from 'react';
...
 import { getNavModel } from 'app/core/selectors/navModel';
 import { getRouteParamsId } from 'app/core/selectors/location';
-
 // Types
-import { StoreState, CoreEvents } from 'app/types/';
+import { CoreEvents, StoreState } from 'app/types/';
 import { UrlQueryMap } from '@grafana/runtime';
-import { DataSourceSettings, DataSourcePluginMeta } from '@grafana/data';
-import { NavModel } from '@grafana/data';
+import { DataSourcePluginMeta, DataSourceSettings, NavModel } from '@grafana/data';
```
Comment (dprokop) [inline public/app/features/datasources/settings/DataSourceSettingsPage.tsx]: 👍

**Review by dprokop — APPROVED** [review]
LGTM 👍

**Pull request merged**